pkg/kubelet/types/constants.go

```diff
@@ -16,8 +16,22 @@ limitations under the License.
 
 package types
 
+import "runtime"
+
 const (
-	PodInfraContainerImage = "gcr.io/google_containers/pause:2.0"
+
 	// system default DNS resolver configuration
 	ResolvConfDefault = "/etc/resolv.conf"
+
+	DefaultPodInfraContainerImageName = "gcr.io/google_containers/pause"
+	DefaultPodInfraContainerImageVersion = "2.0"
 )
+
+// Returns the arch-specific pause image that kubelet use as the default
+func GetDefaultPodInfraContainerImage() string {
+	if runtime.GOARCH == "amd64" {
+		return DefaultPodInfraContainerImageName + ":" + DefaultPodInfraContainerImageVersion
+	} else {
+		return DefaultPodInfraContainerImageName +  "-" + runtime.GOARCH + ":" + DefaultPodInfraContainerImageVersion
```

Comment:
Haha, make sure to have an issue or something if you're planning to change that, so you're listening to the community.

However, the important thing is consistency. We must choose either and go with that all the time. `binary-arch:version` is already implemented in the codebase and for example arm images are pushed with every release with `binary-arch:version` naming, so I don't know if it's worth the effort to change.